apache/superset
# fix: chart validation error not cleared on control value update

### SUMMARY

This fixes another bug introduced by #10113 : when a chart has frontend validator errors (e.g. empty metrics), it's impossible to clear it without refreshing the page.

Also refactored some Cypress tests and did some minor optimization on error collection.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

<img src="https://user-images.githubusercontent.com/335541/86303240-1fcf0f80-bbc0-11ea-9c90-5ce0d922b61f.png" width="300">


### TEST PLAN

Added Cypress tests.

To reproduce the bug:

1. Go to Explore a table chart, clear any metrics under in "Aggregate" mode if there's any
2. Select "Raw" query mode
3. Select some columns
4. Change chart to line chart
5. You should see a validation error on "metrics", add a new metric will not clear the error, preventing users from running the query. However, if they run the query and refresh the page, the query can go through.

### ADDITIONAL INFORMATION
